cabby/model/text/dual_encoder/train.py

```diff
@@ -185,17 +194,35 @@ def train_model(self):
       # Save model and results in checkpoint.
       if self.best_valid_loss > valid_loss:
         self.best_valid_loss = valid_loss
-        util.save_checkpoint(os.path.join(self.file_path, 'model.pt'), 
-          self.model, self.best_valid_loss)
-        util.save_metrics_last_only(
-          os.path.join(self.file_path, 'metrics.tsv'), 
+        util.save_checkpoint(self.model_path, self.model, self.best_valid_loss)
+
+      self.model.train()
+
+
+    logging.info('Finished Training.')
+
+    model_state = util.load_checkpoint(self.model_path, self.model, self.device)
+    valid_loss = model_state['valid_loss']
+
+    logging.info(
+      f'Loaded best model (with validation loss {valid_loss}) for testing.')
+    
+    logging.info('Start testing.')
+
+    test_loss, predictions, true_vals, true_points, pred_points = self.evaluate(
```

Comment:
suggestion: since you have the true and predicted points here for every step, we can compute the distance metrics along with accuracy in the same loop. That way you wont need to run the evaluation code separately.